### DISCUSSION TITLE
Leaving WiFi Manager with button clicks

### DISCUSSION THREAD

**Discussion by coolman111** [arendst/Tasmota#16373] [Ideas]
I noticed after update to Tasmota 12.1.1 that one feature has changed (but I don't know exactly since which release):
When you enter WiFi Manager by clicking fast 6 times the button, in former releases you could leave WiFi Manager with one button click.
Now, one button click only switches on (and off) Relais 1 and the blue LED (on my Gosund SP111 device) indicating WiFi Manager mode stops flashing immediately. But WiFi Manager mode is still going on.
The only way I found to leave WiFi Manager is by clicking the button again 6 times.
My Question:
Is this a bug or the "leaving WiFi Manager" button feature has changed ?
I did not find any information about this in any place.

**Comment by ascillato** [comment]
There is no leaving WiFi Manager button feature. When you press the button it will switch the relay (if configured). To disable the AP mode, just restart the device or wait 3 minutes for the device to restart itself. If the WIFICONFIG parameter is set to 4 (the standard) after the restart, Tasmota will connect to the router (if it is configured).

**Reply by coolman111** [reply]
@ascillato
There was a leaving WiFi Manager button feature, and well documented:
https://tasmota.github.io/docs-7.1/#/Buttons-and-Switches?id=_4-short-presses
"[4 short presses](https://tasmota.github.io/docs-7.1/#/Buttons-and-Switches?id=_4-short-presses)
Start Wi-Fi manager providing an Access Point with IP address 192.168.4.1 and a web server allowing the configuration of Wi-Fi. The LED will blink during the config period. **A single button press during this period will abort and restart the device**."

Well, since Tasmota 8.3, calling the WiFi Manager has changed to 6 short presses, but I'm missing the documentation about how to leave it, if you don't want to wait 3 minutes and you cannot restart the device by other methods.
As I found out by chance, **doing 6 short presses again is the only way to exit WiFi Manager in the current (12.x) version** and booting the device. The reboot button feature has been removed after 8.3.
**IMHO this should be documented in the current docs, as there is no glue about leaving:**
"6 short presses
    Start [WifiConfig 2](https://tasmota.github.io/docs/Commands/#wificonfig). Can be disabled using [SetOption1 1](https://tasmota.github.io/docs/Commands/#setoption1)"

By the way, Adrian, you are wrong about "wificonfig 4 ist standard":
If you set wificonfig 4 and enter WiFi Manager, on exit and restart, wificonfig remains on 2 ! (Tasmota 12.1.1)
So, I have always a rule on my devices:
**rule1 on system#boot do wificonfig 4 endon**
So I can be sure, my devices never will be in the wificonfig 2 state, which makes a lot of problems, when WiFi is weak and unauthorized people could enter the WiFi Manager without having physical access to the device.

So, please confirm:
"Leaving WiFi Manager with again 6 short presses" -- Is this a bug or a feature ?
If it is a feature, I think, it should be documented.

Cheers from Germany
Hans (coolman111)

**Reply by ascillato** [reply]
There is no leaving WiFi Manager feature implemented. The behavior of resetting the device in AP mode when you press the button was eliminated long time ago (we need to fix the docs). That was done so you can actuate devices that are fresh and not connected to any network.

So, if pressing 6 times the button while in wifi manager, restarts the device, that is a "not planned feature". I wouldn't call it a bug since it is useful for you ;)

Since the 6 times button pressing to exit the wifimanager is a really good idea, I will look into that so as to properly implement it. I have to test it, but I think that now it is just crashing and rebooting somehow when calling the AP mode inside the AP mode. That would explain why you loose the wificonfig 4 to wificonfig 2 (in which 4 is the default).

Changing this to a Feature Request.

Thanks. 👍
